core/templates/dev/head/pages/exploration_player/AudioPreloaderService.js

```diff
@@ -20,117 +20,124 @@ oppia.factory('AudioPreloaderService', [
   'AudioTranslationManagerService', 'LanguageUtilService',
+  'ComputeGraphService',
   function($modal, ExplorationContextService, AssetsBackendApiService,
       ExplorationPlayerStateService, UrlInterpolationService,
-      AudioTranslationManagerService, LanguageUtilService) {
-    // List of languages that have been preloaded in the exploration.
-    var _preloadedLanguageCodes = [];
+      AudioTranslationManagerService, LanguageUtilService,
+      ComputeGraphService) {
+    var MAX_NUM_AUDIO_FILES_TO_DOWNLOAD_SIMULTANEOUSLY = 3;
 
-    var _preloadAllAudioFiles = function(languageCode) {
-      var allAudioTranslations =
-        ExplorationPlayerStateService
-          .getExploration().getAllAudioTranslations(languageCode);
+    var _filesBeingDownloaded = [];
+    var _filesToBeDownloaded = [];
+    var _exploration = null;
+    var _isRunning = false;
+    var _audioLoadedCallback = null;
+    var _mostRecentlyRequestedAudioOnCurrentCard = null;
 
-      allAudioTranslations.map(function(audioTranslation) {
-        AssetsBackendApiService.loadAudio(
-          ExplorationContextService.getExplorationId(),
-          audioTranslation.filename);
-      });
-
-      _preloadedLanguageCodes.push(languageCode);
+    var _init = function(exploration) {
+      _exploration = exploration;
     };
 
-    var _showBandwidthConfirmationModal = function(
-        audioTranslationsForContent, languageCode,
-        confirmationCallback) {
-      $modal.open({
-        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-          '/pages/exploration_player/' +
-          'audio_preload_bandwidth_confirmation_modal_directive.html'),
-        resolve: {},
-        backdrop: true,
-        controller: [
-          '$scope', '$modalInstance',
-          'ExplorationPlayerStateService', 'AudioPreloaderService',
-          'LanguageUtilService',
-          function(
-              $scope, $modalInstance,
-              ExplorationPlayerStateService, AudioPreloaderService,
-              LanguageUtilService) {
-            $scope.fileSizeOfCurrentAudioTranslationMB =
-              audioTranslationsForContent[languageCode]
-                .getFileSizeMB().toPrecision(3);
-            $scope.totalFileSizeOfAllAudioTranslationsMB =
-              ExplorationPlayerStateService.getExploration()
-                .getAllAudioTranslationsFileSizeMB(languageCode)
-                .toPrecision(3);
-            $scope.currentLanguageDescription =
-              LanguageUtilService.getAudioLanguageDescription(languageCode);
-            $scope.shouldDownloadAllAudioInExploration = false;
+    var _getAudioFilesInBfsOrder = function(sourceStateName) {
+      var languageCode = AudioTranslationManagerService
+        .getCurrentAudioLanguageCode();
+      var explorationGraph = ComputeGraphService.compute(
+        _exploration.getInitialState().name, _exploration.getStates());
+      var queue = [];
+      var seen = {};
+      var audioFiles = [];
```

Comment:
audioFilenames -- be careful to distinguish between files and filenames, in general.

Reply:
Done